client/src/components/SharedComponents.jsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+const StarRating = ({ rating }) => {
+  const quarteredRating = (Math.round(rating * 4) / 4).toFixed(2);
+  const wholeRating = quarteredRating.slice(0, 1);
+  const remainingFraction = quarteredRating.slice(2);
+  const starFill = [0, 0, 0, 0, 0];
+  for (let i = 0; i < wholeRating; i += 1) {
+    starFill[i] = 100;
+  }
+  starFill[starFill.indexOf(0)] = remainingFraction;
+  return (
+    <>
+      {starFill.map((percent) => <span key={Math.random()} className={`star star-${percent} fa fa-star`} />)}
+    </>
+  );
+};
+const AverageRatings = (ratings) => {
+  let total = 0;
+  let length = 0;
+  for (const key in ratings) {
```

Comment:
We'll probably want to stay away from for-in, as there can be unintended consequences. Basically, it iterates over the whole object, including the prototype, which can cause problems. Object.keys is usually a better solution. It, and some other options, are detailed here: https://www.30secondsofcode.org/articles/s/eslint-refactor-for-in